app/components/elements/TextInput.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Box } from 'rebass/styled-components';
+import { Label, Input } from '@rebass/forms';
+import { Caption } from './FontStyles';
+import { fonts, fontSizes, colors, borders, radii, space } from '../../themes/baseTheme';
+
+export const TextInput = props => {
+  const StyledInput = styled(Input)`
+    border: ${borders.input};
+    box-shadow: none;
+    border-radius: ${radii.input}px;
+    padding: ${styleprops => (styleprops.space ? space[styleprops.space] : 12)}px;
+    caret-color: ${colors.mediumPurple};
+    font-size: ${fontSizes[1]}px;
+    font-family: ${fonts.default};
+    color: ${colors.mediumPurple};
+
+    &::placeholder {
+      color: ${colors.text.primaryTextSubdued};
+    }
+
+    &.active {
+      color: ${colors.text.primaryTextSubdued};
+      box-shadow: none;
+    }
+
+    &:focus {
+      box-shadow: none;
+    }
+
+    &:disabled {
+      background: ${colors.grays[0]};
+      color: ${colors.text.primaryDisabled};
+    }
+  `;
+
+  return (
+    <Box
+      sx={{
+        width: props.width,
+      }}
+    >
+      {props.label &&
+        <Label htmlFor={props.name}>
+          <Caption>{props.label}</Caption>
+        </Label>
+      }
+      <StyledInput id={props.name} {...props} />
```

Comment:
This isn't a big deal, but I'm not sure if you intended to add the `label`, `name`, and `width` props to the `StyledInput` as well.  I don't think they could really cause issues (though I'm wondering if the width being added to the padding applied to the input could be...)

I'd consider pulling the other props out and getting the input props via rest destructuring (unless you intended to pass all the props through)

```
const { label, name, width, ...inputProps } = props;
...
<StyledInput id={props.name} {...inputProps} />
```

Reply:
Not entirely intentional, but I didn't see it hurting so I shrugged it off and didn't come back to address it after time of writing. Destructured is much nicer/cleaner and more specific/intentional. 👍